atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -215,14 +185,87 @@ def _send_mail(self, receivers_list, subject, body):
+                raise RuntimeError("Empty email_domain specified")
+            return '@'.join([obj['owner_name'], self.email_domain])
+
+    def _get_koji_submitter(self):
+        koji_task_id = self.metadata['labels']['koji-task-id']
+        koji_task_info = self.session.getTaskInfo(koji_task_id)
+        koji_task_owner = self.session.getUser(koji_task_info['owner'])
+        koji_task_owner_email = self._get_email_from_koji_obj(koji_task_owner)
+        self.submitter = koji_task_owner_email
+        return koji_task_owner_email
+
+    def _get_koji_owners(self):
+        result = []
+        koji_build_id = self.metadata['labels']['koji-build-id']
+        koji_build_info = self.session.getBuild(koji_build_id)
+
+        koji_tags = self.session.listTags(koji_build_id)
+        for koji_tag in koji_tags:
+            koji_tag_id = koji_tag['id']
+            koji_package_id = koji_build_info['package_id']
+            koji_pkg_tag_config = self.session.getPackageConfig(koji_tag_id, koji_package_id)
+            koji_pkg_tag_owner = self.session.getUser(koji_pkg_tag_config['owner'])
+
+            result.append(self._get_email_from_koji_obj(koji_pkg_tag_owner))
+
+        return result
+
+    def _get_logs_url_from_build(self):
+        # We're importing this here in order to trap ImportError
+        from koji import PathInfo
+        pathinfo = PathInfo(topdir=self.koji_root)
+
+        koji_task_id = self.metadata['labels']['koji-task-id']
+        koji_task_output = self.session.listTaskOutput(koji_task_id)
+        koji_logs = [filename for filename in koji_task_output if filename.endswith('.log')]
+        files_dir = '%s/%s' % (pathinfo.work(), pathinfo.taskrelpath(koji_task_id))
+        if "openshift-final.log" not in koji_logs:
+            raise RuntimeError("Openshift logs was not found in koji task outputs")
+
+        return "/".join([files_dir, "openshift-final.log"])
+
+    def _get_receivers_list(self):
+        receivers_list = []
+        if self.additional_address:
+            receivers_list.append(self.additional_address)
+
+        if self.session and (self.to_koji_submitter or self.to_koji_pkgowner):
+            if self.to_koji_submitter:
+                try:
+                    koji_task_owner_email = self._get_koji_submitter()
+                except Exception:
+                    self.log.exception("Failed to include a task submitter")
+                else:
+                    receivers_list.append(koji_task_owner_email)
+
+            if self.to_koji_pkgowner:
+                try:
+                    koji_task_owner_emails = self._get_koji_owners()
+                except Exception:
+                    self.log.exception("Failed to include a package owner")
+                else:
+                    receivers_list += koji_task_owner_emails
+
+        if not receivers_list:
+            raise RuntimeError("No recepients found")
+
+        return list(chain(receivers_list))
```

Comment:
I'm not sure this will do what you want:
```python
>>> list(chain(['spam', ['eggs', 'sausage'], 'bacon']))
['spam', ['eggs', 'sausage'], 'bacon']

# Docs indicate arg must be expanded (use star)
>>> list(chain(*[['eggs', 'sausage'], 'bacon']))
['eggs', 'sausage', 'b', 'a', 'c', 'o', 'n']
```
To work with `chain`, or `chain.from_iterable`, each list element should be a list, which is not our case.
Here's a hackish solution:

```python
from types import GeneratorType
breakfast = [['eggs', 'sausage'], 'bacon']

def force_list(item):
    if type(item) not in (list, tuple, GeneratorType):
        item = [item]
    return item

list(chain.from_iterable(force_list(item) for item in breakfast))
```